### ISSUE TITLE
frozendict repr error w/ Metabolite entries

### ISSUE THREAD

**Issue by pstjohn** [opencobra/cobrapy#502]
In dubugging reaction.py (#503), I got an error when I tried to print out `old_metabolites`, a frozendict. Error message blow, looks like this was introduced with the new YAML round-tripping (#496). Pinging @Midnighter, do we need the sort to make the yaml output deterministic? Should we have a separate FrozenOrderedDict vs. FrozenDict? (why do we have frozendicts? 😄 )

```python
In [3]: old_coefficients
Out[3]: ---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Users/pstjohn/anaconda3/envs/cobra/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj)
    670                 type_pprinters=self.type_printers,
    671                 deferred_pprinters=self.deferred_printers)
--> 672             printer.pretty(obj)
    673             printer.flush()
    674             return stream.getvalue()

/Users/pstjohn/anaconda3/envs/cobra/lib/python3.6/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    381                             if callable(meth):
    382                                 return meth(obj, self, cycle)
--> 383             return _default_pprint(obj, self, cycle)
    384         finally:
    385             self.end_group()

/Users/pstjohn/anaconda3/envs/cobra/lib/python3.6/site-packages/IPython/lib/pretty.py in _default_pprint(obj, p, cycle)
    501     if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
    502         # A user-provided repr. Find newlines and replace them with p.break_()
--> 503         _repr_pprint(obj, p, cycle)
    504         return
    505     p.begin_group(1, '<')

/Users/pstjohn/anaconda3/envs/cobra/lib/python3.6/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    699     """A pprint that just redirects to the normal repr function."""
    700     # Find newlines and replace them with p.break_()
--> 701     output = repr(obj)
    702     for idx,output_line in enumerate(output.splitlines()):
    703         if idx:

/Users/pstjohn/Packages/cobrapy/cobra/util/util.py in __repr__(self)
     36     def __repr__(self):
     37         if self.__items is None:
---> 38             self.__items = sorted(iteritems(self.__dict))
     39         return '{}({!r})'.format(self.__class__.__name__, self.__items)
     40

TypeError: '<' not supported between instances of 'Metabolite' and 'Metabolite'
```

**Comment by Midnighter** [comment]
Yes, my mistake. (Or lack of tests :stuck_out_tongue: ) I changed `FrozenDict`. I've seen it in one place only so far which is to prevent direct changes when you get the `Reaction.metabolites`.

**Comment by Midnighter** [comment]
I suppose a quick solution would be to scrap `FrozenDict` and simply return a shallow copy.

**Comment by pstjohn** [comment]
Seems like a lot of code just for that reason, I'd be happy with a copy. Wouldn't be any slower than the frozendict -- looks like there was a PEP for unmutable dicts that got rejected.

Would we then need to use OrderedDicts in the yaml export?

**Comment by Midnighter** [comment]
> Would we then need to use OrderedDicts in the yaml export?

Actually, I used `OrderedDict` in `cobra.io.dict` a lot and `Reaction.metabolites` is the only part that I had to sort by ID for consistent results.

**Comment by hredestig** [comment]
fixed by #504
